### PR TITLE
fix: update Map event listeners on prop changes

### LINKED ISSUE #324: Map Component's onClick props do not update
**Describe the bug**
When using the onClick function of the map component props dont get updated and instead remain with the first initial value when the component mounts. This only happens inside the onClick function.

**To Reproduce**
1. Setup the map
2. add onClick function
3. P

**Expected behavior**
I expect that props inside the onClick function get the current value its getting from its parent

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Chrome
 - Version 90.0.4430.85 (Official Build) (x86_64)

**Additional context**
https://codesandbox.io/s/886nn
Made a simple sandbox showing the error

### PR DESCRIPTION
**Pull request by chwallen**
When updating the event listener props, the MapGL component does not propagate these changes to the underlying Mapbox instance. This makes it impossible to use state inside, for example, the `onClick` event handler as it will be bound to the first render. It also makes it impossible to remove event handlers dynamically.

This PR updates any event listeners that has changed in a similar way to how `componentDidMount` applies the initial listeners.

Fixes #324

### CODE REVIEW

File: src/components/MapGL/index.js [hunk 1]
```diff
@@ -443,6 +444,37 @@ class MapGL extends PureComponent<Props, State> {
...
+   * @private
+   * @param {Props} prevProps
+   * @param {Props} newProps
+   */
+  _updateMapEventListeners(prevProps: Props, newProps: Props): void {
+    events.forEach((event) => {
+      const propName = `on${capitalizeFirstLetter(event)}`;
+      const prevEventHandler = prevProps[propName];
+      const newEventHandler = newProps[propName];
+      if (prevEventHandler !== newEventHandler) {
```
Comment (stepankuzmin) [inline src/components/MapGL/index.js:459]: Could you please use `isArraysEqual` for comparing these props? https://github.com/urbica/react-map-gl/blob/12906655d76256731d4020cbc3883a7996c7463e/src/utils/isArraysEqual.js#L1

**Review by stepankuzmin — CHANGES_REQUESTED** [review]
Hi @chwallen,

Thanks for the contribution! Sorry for the delayed response

File: src/components/MapGL/index.js (hunk 1 above)
Comment (chwallen) [reply src/components/MapGL/index.js:459]: I'm guessing something like this?

```js
if (!(prevEventHandler === newEventHandler ||                                                                                                                                                                                                                                                                   
    (Array.isArray(prevEventHandler) &&
     Array.isArray(newEventHandler) &&
     isArraysEqual(prevEventHandler, newEventHandler)))) {
```

No worries on the delay!

Comment (stepankuzmin) [reply src/components/MapGL/index.js:459]: Yep, but with array comparison first 👍

Comment (chwallen) [reply src/components/MapGL/index.js]: Done!

**Comment by stepankuzmin** [comment]
Thanks for the contribution, @chwallen!

**Pull request merged**